da/celestia/celestia.go

```diff
@@ -133,48 +134,55 @@ func (c *DataAvailabilityLayerClient) Start() (err error) {
 		return nil
 	}
 
-	rpc, err := openrpc.NewClient(c.ctx, c.config.BaseURL, c.config.AuthToken)
+	var rpc *openrpc.Client
+	rpc, err = openrpc.NewClient(c.ctx, c.config.BaseURL, c.config.AuthToken)
 	if err != nil {
 		return err
 	}
 
-	state, err := rpc.Header.SyncState(c.ctx)
-	if err != nil {
-		return err
-	}
-
-	if !state.Finished() {
-		c.logger.Info("waiting for celestia-node to finish syncing", "height", state.Height, "target", state.ToHeight)
-
-		done := make(chan error, 1)
-		go func() {
-			done <- rpc.Header.SyncWait(c.ctx)
-		}()
-
-		ticker := time.NewTicker(1 * time.Minute)
-		defer ticker.Stop()
-
-		for {
-			select {
-			case err := <-done:
-				if err != nil {
-					return err
-				}
-				return nil
-			case <-ticker.C:
-				state, err := rpc.Header.SyncState(c.ctx)
-				if err != nil {
+	go func() {
+		sync := func() error {
+			done := make(chan error, 1)
+			go func() {
+				done <- rpc.Header.SyncWait(c.ctx)
+			}()
+			ticker := time.NewTicker(1 * time.Minute)
+			defer ticker.Stop()
+			for {
+				select {
+				case err := <-done:
+					if err != nil {
+						c.logger.Error("Failed to sync Celestia DA.", "Error", err)
+					}
 					return err
+				case <-ticker.C:
+					state, err := rpc.Header.SyncState(c.ctx)
+					if err != nil {
+						c.logger.Error("Failed to sync Celestia DA.", "Error", err)
+						return err
+					}
+					if !state.Finished() {
+						c.logger.Info("celestia-node still syncing", "height", state.Height, "target", state.ToHeight)
+					}
 				}
```

Comment:
missing return here? in case `state.Finished()` is true

Reply:
i think its fine not to return here, since in case there is no error the ticker is just used for logging the syncing progress. in case it finishes it will write to done channel by rpc.Header.SyncWait func and it will exit, or retry in case of error.
however, i realized there is a bug, finalizing before executing the retry function. i fixed it.

Reply:
so why checking `state.Finished()` at all? by your logic we could just log "sync in progress" on timer tick.
I guess we wanted a double check on the `SyncWait` signal